.github/workflows/cargo-test.yml

```diff
@@ -101,6 +101,10 @@ jobs:
     timeout-minutes: 90
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: ${{ inputs.fetch-depth }}
+          ref: ${{ inputs.ref }}
+          repository: ${{ inputs.repository }}
```

Comment:
Why do we need to do a special checkout for benchmarks?
Shouldn't all the tests check out the latest code? For example the test workflow on the same file doesn't do the same checkout as the benchmarks although they both run on the same PR

Reply:
look at line 63